### ISSUE TITLE
Custom art sometimes does not display

### ISSUE THREAD

**Issue by rajanizara** [jojo2357/Music-Discord-Rich-Presence#40]
Hello, I have a really strange bug. Some of the covers I uploaded on my own application won't display.
I followed correctly the tutorial and most of my albums are displayed with their cover in Discord, however a few don't and I can't fix that.
The .dat file and the Discord application have the same asset name, the .dat file and the track tag have the same title BUT the cover doesn't display simply
Among the 60 albums I registered, I estimate the bugging album art covers around 10

See one of the culprits on the screenshots below :
![tag](https://user-images.githubusercontent.com/53912802/119265497-50100a00-bbe7-11eb-8207-ec482039e00d.png)
![ddd](https://user-images.githubusercontent.com/53912802/119265501-50a8a080-bbe7-11eb-8b52-84d9b1282e73.png)
![vscode](https://user-images.githubusercontent.com/53912802/119265502-51413700-bbe7-11eb-91ef-b9ae233574c0.png)
![discord](https://user-images.githubusercontent.com/53912802/119265661-c4e34400-bbe7-11eb-8222-9be02068f1f9.png)

Most of bugging albums are however displayed as below 👍 
![discord2](https://user-images.githubusercontent.com/53912802/119265686-daf10480-bbe7-11eb-8754-cc9f0298d4a5.png)

**Comment by jojo2357** [comment]
Can I see the `.dat` file that you used and what version it says you are running? 

The first bugged image where nothing shows up means that the key in the data file and the actual asset name in discord do not match or that you just uploaded the art and discord is processing it.

The second image where it shows the bees could mean that the current album playing and what you have saved as a key do not match so it defaults to my application.

More info is needed.

**Comment by rajanizara** [comment]
Here is my .dat file :
https://drive.google.com/file/d/1hqxlr1I5UkGQ9e-HFYTwtpyXjp04-y1k/view?usp=sharing
I didn't see any «version» mentionned in it.
However, for the Yann Tiersen album named **Le Fabuleux Destin d’Amélie Poulain** I'm certain that the names in assets and.dat files are the same (I reverified just now) and I uploaded the cover more than 10 minutes ago when I screenshoted.
For the a-ha album named **MTV Unplugged (Summer Solstice)**, the only difference between its name and the .dat key is that I replaced spaces by underscores and deleted parenthesis.
Maybe you would be added to my Discord app as a tester ?

**Comment by RoguedBear** [comment]
I've had this issue happen with me too multiple times when i manually add album arts.
When discord shows the my rich presence as in the first screenshot [no image; just text], and the MDRP console says `this is a good one, check your DRP!` in green text i assume that everything is done right and that Discord just needs time to process the image or something.

A little later, the album art does end up showing eventually.

For the 2nd screenshot, i just check what MDRP console has to say and follow from there.
it can either be a yellow `Artist do no match` or a red `no key found`

You can see what the MDRP console says in your case.

**Comment by Smaltin** [comment]
I just want to note that "Various Artists" doesn't work, that's just Spotify's way of saying multiple people. If you want it to work, I'd suggest putting in every artist that worked on that song.

**Comment by jojo2357** [comment]
Or, alternatively, you can just leave off artists. If the mdrp console says something like `album keyed for the wrong artist` like rougedbear mentioned, then you can leave off the artist names. It is just a tool I provided so that if u have duplicate album names that you could still hope to separate the albums and get the right arts

**Comment by jojo2357** [comment]
Doh! Just realized that discord has a 32 character limit on the album keys and when I overhauled the albums to include artists, the overcharacter warning went away. I can make a new release to fix it.

**Comment by rajanizara** [comment]
> Doh! Just realized that discord has a 32 character limit on the album keys

I can confirm that all albums which were not displaying the album cover -as below- had a data key too long :
![discord](https://user-images.githubusercontent.com/53912802/119265661-c4e34400-bbe7-11eb-8222-9be02068f1f9.png)
So I modified them and now it works ! All of my bugs disappeared ! (also I had to correct some album titles due to case sensitivity)

>I just want to note that "Various Artists" doesn't work
>Or, alternatively, you can just leave off artists.

Because of my previous bugs I couldn't say if it is a problem. However, so as to take 0 risks I deleted all artists in my data file. It wasn't problematic thanks to the absence of multiple albums named the same way.

Thanks for helping me !

> I can make a new release to fix it.

That would be welcomed
